### DISCUSSION TITLE
Vite ssrContext does not exist on type

### DISCUSSION THREAD

**Discussion by Abner-Corona** [quasarframework/quasar#13059] [General - Components / Directives / etc]
Hello, i have a issue when create a new project with vite and typescript, the ssrContext not found in types, i selec t pinia but in vuex doesnt work too,  in webpack works fine

<img width="550" alt="image" src="https://user-images.githubusercontent.com/51718510/161594818-68b58ae2-8df5-43c0-9800-dabb157d3dcd.png">
<img width="610" alt="image" src="https://user-images.githubusercontent.com/51718510/161594431-2869cdee-b64c-480b-bf01-b7a4d6aaa801.png">
<img width="515" alt="image" src="https://user-images.githubusercontent.com/51718510/161594500-efe41b2a-a3a7-4096-9e95-6f59a1d942f9.png">

**Comment by rstoenescu** [comment]
ssrContext is available only for SSR mode. You first need to add that mode.

**Reply by Abner-Corona** [reply]
> ssrContext is available only for SSR mode. You first need to add that mode.

I added the ssr mode but dont fix the lint, in webpack on ctrl click on ssrContext sent me to ssr.d.ts, in vite dont, the project runs fine only the lint maybe my  extensions

<img width="599" alt="image" src="https://user-images.githubusercontent.com/51718510/161597551-58a83f97-1dff-454d-9304-0128af383a00.png">
<img width="495" alt="image" src="https://user-images.githubusercontent.com/51718510/161597309-8ea0b255-0469-4c1c-ac35-fc0f16afb915.png">

**Reply by rstoenescu** [reply]
Just to be sure there's no inconsistency or anything, can you restart VSCode and tell me if it is still an issue pls?

**Reply by Abner-Corona** [reply]
> 
I remove complety visual studio and reinstall it with 0 extensions and default settings, and still the lint
![image](https://user-images.githubusercontent.com/51718510/161603767-932f1528-4e37-4446-8f94-74cfb1e9d524.png)

**Comment by bekaku** [comment]
same issue

![This is an image](https://www.linkpicture.com/q/Untitled_197.png)

**Comment by Grawl** [comment]
Same for me. Just installed Quasar with Vite using `yarn create quasar` and there's no `ssrContext` in `boot()` callback params.

The problem is that `tsconfig.json` generared by Quasar Wizard contains these lines:

```json
"include": [
    "src/**/*.ts",
    "src/**/*.d.ts",
    "src/**/*.tsx",
    "src/**/*.vue"
  ]
```

This prevents `src-ssr/ssr-flat.d.ts` coming into account of TypeScript.

I remove that and `ssrContext` is now defined.

Also, this will work if you add `"src-ssr/*.d.ts"` to that `include`.

**Reply by tisso023** [reply]
I'm currently upgrading to app-vite 2.2 (from 1.8) and have tried everything to get the ssrContext. I've now googled it properly and tried your suggestion. Thank you very much! It works now...finally! ;-)

**Reply by yusufkandemir** [reply]
@tisso023 yes, the [app-vite v2 upgrade guide](https://quasar.dev/quasar-cli-vite/upgrade-guide#beginning-of-the-upgrade-process) covers this:
![image](https://github.com/user-attachments/assets/56196d48-be71-4aee-ab62-585f544d348a)
